mono/mini/interp/transform.c

```diff
@@ -4525,18 +4521,27 @@ mono_interp_transform_method (InterpMethod *imethod, ThreadContext *context, Mon
 	generate (method, header, imethod, is_bb_start, generic_context, &error);
 
 	mono_metadata_free_mh (header);
+	g_free (is_bb_start);
 
 	if (!mono_error_ok (&error)) {
 		mono_os_mutex_unlock (&calc_section);
```

Comment:
you forget to remove a couple of unlocks, like this for example (that's why the mcs tests aren't passing)

Reply:
Fixed.